lib/auth/bot_test.go

```diff
@@ -575,9 +575,11 @@ func TestRegisterBotCertificateGenerationStolen(t *testing.T) {
 	require.NoError(t, err)
 	require.NotEmpty(t, locks)
 
-	// The original client should now be locked out.
-	_, err = renewedClient.Ping(ctx)
-	require.ErrorContains(t, err, "access denied")
+	// The original client should be locked out.
+	require.Eventually(t, func() bool {
```

Comment:
Wrapping this in a `require.Eventually()` isn't exactly my first choice but I wasn't able to reproduce the failure locally in several thousand iterations with the race detector enabled. I expect this is just some cache sync or other acceptable raciness but would be happy to hear alternative solutions!